sdc_client/async_api_client.py

```diff
@@ -0,0 +1,131 @@
+import asyncio
+import aiohttp
+
+from sdc_client.base_api_client import _BaseStreamSetsApiClient, UnauthorizedException
+from sdc_client.interfaces import IStreamSets
+
+
+MAX_TRIES = 3
+
+
+def async_endpoint(func):
+    """
+    logs errors and returns json response
+    """
+    async def wrap(*args, **kwargs):
+        async with args[0].session:
+            for i in range(MAX_TRIES):
+                try:
+                    res = await func(*args, **kwargs)
+                    res.raise_for_status()
+                    if res.text:
+                        return await res.json()
+                except aiohttp.ClientConnectionError:
+                    if i == MAX_TRIES - 1:
+                        raise
+                    asyncio.sleep(2 ** i)
+                    continue
+                except aiohttp.ClientResponseError:
+                    if res.status == 401:
+                        raise UnauthorizedException('Unauthorized')
+                    # TODO parse error responce
```

Comment:
Could we move **_parse_error_response()** to base class and use it here?

Reply:
What do you mean? Base class doesn't work with specific lib, but in this case we need to use aiohttp. Do you want to extract this code to method, like in sdc_client/api_client.py?

Reply:
I mean what do you plan to do with #TODO?:)